lua/lspconfig/ui/lspinfo.lua

```diff
@@ -214,24 +260,53 @@ return function()
     end
   end
 
-  local matching_config_header = {
-    '',
-    'Configured servers list: ' .. table.concat(vim.tbl_keys(configs), ', '),
+  local configured_head = 'Configured servers list: '
+  local start_row = #buf_lines
+  local servers_ctx = {
+    servers = vim.tbl_keys(configs),
+    width = floating_width,
+    header = configured_head,
   }
-  vim.list_extend(buf_lines, matching_config_header)
+  local servers_list, hi_scope = generate_servers_list(servers_ctx)
+
+  vim.list_extend(buf_lines, servers_list)
 
   local fmt_buf_lines = indent_lines(buf_lines, ' ')
 
   fmt_buf_lines = vim.lsp.util._trim(fmt_buf_lines, {})
 
-  vim.api.nvim_buf_set_lines(bufnr, 0, -1, true, fmt_buf_lines)
-  vim.api.nvim_buf_set_option(bufnr, 'modifiable', false)
-  vim.api.nvim_buf_set_option(bufnr, 'filetype', 'lspinfo')
+  api.nvim_buf_set_lines(bufnr, 0, -1, true, fmt_buf_lines)
+  api.nvim_buf_set_option(bufnr, 'modifiable', false)
+  api.nvim_buf_set_option(bufnr, 'filetype', 'lspinfo')
 
-  vim.api.nvim_buf_set_keymap(bufnr, 'n', '<esc>', '<cmd>bd<CR>', { noremap = true })
-  vim.api.nvim_command(
-    string.format('autocmd BufHidden,BufLeave <buffer> ++once lua pcall(vim.api.nvim_win_close, %d, true)', win_id)
-  )
+  api.nvim_buf_attach(bufnr, false, {
+    on_detach = function()
+      if api.nvim_win_is_valid(win_id) then
+        api.nvim_win_close(win_id, true)
+      end
+    end,
+  })
+
+  vim.keymap.set('n', '<ESC>', function()
+    if api.nvim_buf_is_valid(bufnr) then
+      api.nvim_buf_delete(bufnr, { force = true })
+    end
+  end, { buffer = bufnr, nowait = true })
+
+  local au_id
+  au_id = api.nvim_create_autocmd({ 'BufLeave,BufHidden' }, {
```

Comment:
Unless I'm missing something, you probably want to separate these?

```suggestion
  au_id = api.nvim_create_autocmd({ 'BufLeave', 'BufHidden' }, {
```